code-challenges.js

```diff
@@ -21,48 +21,113 @@ const fruit2 = "banana"
+// Convert the data to an Array. I can do this with the .split() method.
+// Use the .reverse() method to reverse the locations of each of the elements.
+// Use the .join() method to convert the array back into a string.
+// Wrap the code into a console.log() so that it's visible in the console.
 
-const currentCohort = "Foxtrot 2022"
+const currentCohort = "Foxtrot 2022";
 // Expected output: "2202 tortxoF"
 
+console.log(currentCohort.split("").reverse().join(""));
+
+//----------- AFTER RUNNING -----------
+// The output I got was "2202 tortxoF", which matches the expected output!
+
 // --------------------4) Create the code that will return the last index of the given value from the array using the test variables provided below.
 
 // Describe your process:
+// To access the last index of any given element, I'll need to use the method .lastIndexOf(). This method will return the LAST index of any given value that I call it for.
+// I'll use a console.log() with the variable the array is stored in with the .lastIndexOf() method in order to check that the output matches what I'm expecting it to for this assignment.
 
-const myNumbers = [13, 34, 42, 5, 5, 10, 27, 42, 10]
-const givenValue1 = 42
+const myNumbers = [13, 34, 42, 5, 5, 10, 27, 42, 10];
+const givenValue1 = 42;
 // Expected output: 7
 
-const givenValue2 = 10
+const givenValue2 = 10;
 // Expected output: 8
 
+console.log(myNumbers.lastIndexOf(givenValue1));
+
+//----------- AFTER RUNNING -----------
+// The output I got for givenValue1 was 7 as expected, and the output I got for givenValue2 was 8 as expected. Code seems to be working! :D
+
 // --------------------5) Create the code that will sort all the numbers in an array from largest to smallest. Use the test variables provided below.
 
 // Describe your process:
+// This one is definitely a little confusing for me as far as methods go. After re-reading through the syllabus, I found the .sort() function which I believe will work well for this purpose. I'll need to console.log() the variable holding the array, then apply .sort() to it.
+// In the given example on the syllabus, the function is looking for smallest to largest - ((a, b) => a - b).
+// This function will take two arguments, a & b.
+// Then, the function will subtract value b from value a.
 
-const sanDiegoSummerTemperatures = [79, 80, 72, 73, 82, 77, 76]
+// I had to do a little research to understand this one. Here's what I learned:
+// The .sort() method cannot automatically work on numbers. You have to give it an argument so that the logic will understand what result you want.
+// When the function runs, it's going to return a positive, negative, or 0 value.
+// As the function continues to subtract numbers from each other, it's going to order them from either negative or positive numbers first depending on what argument you pass it. For our case, I wanted it to start with the negative integers, so I placed the second argument first in the function.
```

Comment:
This is an incredible explanation on how this is working. This one is a little tricky since we have not learned about something called higher order functions which is essentially what we are using here to sort the numbers properly.  I think this will make a little more sense tomorrow when we dive a little deeper into higher order functions.